### DISCUSSION TITLE
Ability to detect when the branch changes

### DISCUSSION THREAD

**Discussion by StevenRasmussen** [VsixCommunity/Community.VisualStudio.Toolkit#71] [Q&A]
Does anyone know if there is a way to detect when the source control branch is changed with a solution already opened?  I know that I can subscribe to the `SolutionEvents.Opened` event, but as I understand it this is not always fired when a branch changes as I believe there are some optimizations around whether to reload a solution/project.

**Comment by yannduran** [comment]
Wouldn't that be controlled by the source code provider, not VS itself?

**Reply by StevenRasmussen** [reply]
True.  However, there are some abilities in the SDK that allow generic communication with a source code provider in the `DTE2.SourceControl` class but nothing with regards to branches.  Just wanted to see if anyone knew of some interface or method/event that I might have missed that would help out with this.

**Reply by yannduran** [reply]
Fair enough. Sorry that I can't be of any help.
